### ISSUE TITLE
Determine a logical branch strategy that unblocks both engineering and content teams

### ISSUE THREAD

**Issue by jpvajda** [newrelic/docs-website#1237]
We need to come up with a way to prevent having to "coordinate" releases across teams as both content and engineering should be de-coupled from releasing their changes.

## Current State

We all work from the develop branch and periodically push to main to cut a release. This worked ok when only the engineering team was making changes, but now with content changes being staged there it's becoming a coordination problem.

## Solution

stashng content on a different branch would mean that fixes can free go out without worry that we are going to deploy some content that was intended to be staged for a different time. So let's come up with a logical branch strategy that let's the content team work unfettered and the engineering team work unfettered.

**Comment by jpvajda** [comment]
It's probably best to discuss the best approach with the content-team and decide an approach.  

1. change process for merging into develop, and assume that develop is always ready to deploy
2. use a content-staging branch that is just for content. 

> need to discuss the workflow..as it requires some process to ensure that branch stays current.

**Comment by jpvajda** [comment]
We prefer 1. as Develop should be "ready to ship".

**Comment by jpvajda** [comment]
@austin-schaefer just need input into the approach we came up with.

**Comment by austin-schaefer** [comment]
@jpvajda What's the difference between `1.` and the current state? Barring that accidental bad commit earlier this week, my assumption is that anything merged into develop is ready to deploy.

**Comment by mmfred** [comment]
It would be great to know more about the problem of potentially releasing content that's not quite ready. It's my understanding that no PR will be released until it's merged (in a PR from `develop` to `main`). I did erroneously merge a PR yesterday where the user hadn't signed the CLA, and I'm sure that was frustrating for all, and particularly for the engineers whose fixes were held up while I tried to solve for this. But that was (I hope) an anomaly.

The issue I have seen since we started regular merges last week is that the checks are very slow, so any PRs that get merged to `develop` before the PR from `develop` to `main` is merged are then pulled into that `develop` to `main` PR. This then restarts the checks, keeping the `develop` to `main` PR open longer, allowing more PRs to get added, and it all bogs down. So it would be great to have a solution that solves for that. Maybe having docs work on another branch would help, but it does raise the likelihood of merge conflicts.

**Comment by jpvajda** [comment]
@austin-schaefer  1. would mean your team would work under the assumption that pushing to develop means that work is ready to go live at any time.

**Comment by jpvajda** [comment]
@mmfred so far, the first occurrence of having to coordinate a release was yesterday so perhaps to your point this was a one time occurrence. If you currently take the approach to:

1. work off a seperate branch
2. keep that branch up to date by  `git pull origin develop` 
3. Resolve any merge conflicts
4. then merge your branch into develop via a PR.
5. Develop is consider "ready to ship" 

We should be fine. Of course when you need to `git revert` any commit from develop, it's going to hold things up. 😄  But ideally that's not happening all that often. 

The idea really is: when merging to develop all code should be considered eady to be deployed to production.

**Comment by mmfred** [comment]
@jpvajda - yes. Agree! I think we're doing pretty well on that workflow. (except for that revert yesterday).

**Comment by jpvajda** [comment]
@mmfred awesome. I don't see this level of detail in the contributor guide:

https://github.com/newrelic/docs-website/blob/develop/CONTRIBUTING.md#using-the-develop-branch

So I'll add it.

**Comment by jerelmiller** [comment]
One thing we can do to prevent the CLA thing from happening in the future is to make it a required status check. Then GitHub will prevent us from merging PRs unless the CLA is signed, therefore preventing future accidents.
